Workflow/Sources/AnyWorkflowConvertible.swift

```diff
@@ -147,3 +147,18 @@ struct OutputBlockingWorkflow<Child: AnyWorkflowConvertible>: Workflow {
             .rendered(in: context)
     }
 }
+
+// MARK: -
+
+extension AnyWorkflowConvertible where Output == Never {
+    /// A convenience for workflows that don't produce output but want to change the declared output type.
+    ///
+    /// This equivalent to `.mapOutput { switch $0 {} }` but avoids the compiler warning for unreachable code in the provided transform.
+    public func mapOutput<NewOutput>() -> AnyWorkflow<Rendering, NewOutput> {
+        asAnyWorkflow().mapOutput(silenceUnreachableWarning(_:))
```

Comment:
nit: i think this could just call through to the `mapOutput` implementation without doing the `asAnyWorkflow()` call (since that will happen in the version that takes a transform parameter).